pkg/state/types.go

```diff
@@ -552,6 +555,49 @@ func (p1 *Plugin) EqualWithOpts(p2 *Plugin, ignoreID,
 	return reflect.DeepEqual(p1Copy, p2Copy)
 }
 
+// Helper function to sort a slice of interface{}
+func sortInterfaceSlice(slice []interface{}) []interface{} {
+	// Convert []interface{} to []string for sorting
+	strSlice := make([]string, len(slice))
+	for i, v := range slice {
+		strSlice[i] = fmt.Sprintf("%v", v)
+	}
+
+	sort.Strings(strSlice)
+
+	// Convert back to []interface{}
+	sortedSlice := make([]interface{}, len(strSlice))
+	for i, v := range strSlice {
+		sortedSlice[i] = v
+	}
+
+	return sortedSlice
+}
```

Comment:
How about using a type that can be sorted and thus not allocating unnecessarily?

Something like this: d7c9b78036793944648cb0a842000018d0635975

WDYT?

Reply:
Pushed it directly to this PR: https://github.com/Kong/go-database-reconciler/pull/93/commits/d7c9b78036793944648cb0a842000018d0635975.

PTAL